src/Ztm.Zcoin.NBitcoin/Exodus/TransactionEncoder.cs

```diff
@@ -24,6 +24,34 @@ public TransactionEncoder(IEnumerable<ITransactionPayloadEncoder> encoders)
             this.encoders = encoders.ToDictionary(e => e.Type);
         }
 
+        public byte[] Encode(ExodusTransaction transaction)
+        {
+            if (transaction == null)
+            {
+                throw new ArgumentNullException(nameof(transaction));
+            }
+
+            using (var stream = new MemoryStream())
+            using (var writer = new BinaryWriter(stream, Encoding.UTF8))
+            {
+                var type = transaction.Type;
+
+                writer.Write(IPAddress.HostToNetworkOrder((short)transaction.Version));
+                writer.Write(IPAddress.HostToNetworkOrder((short)transaction.Type));
+
+                if (!this.encoders.TryGetValue(type, out var encoder))
+                {
+                    throw new TransactionFieldException(
```

Comment:
We should use `ArgumentException` instead. `TransactionFieldException` was designed to catch invalid exodus payload.